sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -56,13 +57,26 @@ case class JdbcType(databaseTypeDefinition : String, jdbcNullType : Int)
  */
 @DeveloperApi
 abstract class JdbcDialect extends Serializable {
+  /**
+   * The tag of the database, it must be lowercase, the subsequent will match
+   * the corresponding dialect through this tag, so it must be strictly one-to-one
+   * correspondence with jdbc url,such as:
+   * mysql  url:  jdbc:mysql://localhost/db  dbTag: mysql
+   * oracle url:  jdbc:oracle://localhost/db dbTag: oracle
+   * db2    url:  jdbc:db2://localhost/db    dbTag: db2
+   * ......
+   */
+  def dbTag: String
```

Comment:
Hm, the problem here is that this will break any custom dialects as it adds a new required method to implement. I am not sure it's worth it, even though we can possibly introduce such breaking changes. Part of the reason I don't think it's worth it is just that a non-lowercase string is pretty nonstandard.

I would either push the lowercase down into each implementation, or provide a default implementation that returns nothing. That at least covers existing implementations which will already override `canHandle`.

Reply:
So how do you look like this, the canHandle in each implementation is cover to lowercase
```
private case object MySQLDialect extends JdbcDialect {

  override def canHandle(url : String): Boolean =
      url.toLowerCase(Locale.ROOT).startsWith("jdbc:mysql")
```

Reply:
Not lowercase is not standard, but the uppercase url can read data correctly from mysql, I think it needs to be compatible here.

Reply:
Yes, at best that is what I'd do here. It's not worth breaking the developer API even in a minor way.

Reply:
Yea, that new one looks nice to me, too.